### ISSUE TITLE
java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkState(ZLjava/lang/String;Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)V

### ISSUE THREAD

**Issue by drno-reg** [SeleniumHQ/selenium#3880]
Hello,

I have browser Google Chrome 57.0.2987.133 (64-bit), webdriver chrome 
ChromeDriver 2.29.461591 (62ebf098771772160f391d75e589dc567915b233), Java
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode).

Tried to run Code on Java

```
        System.setProperty("webdriver.chrome.driver", "C:\\Server\\templates\\selenium\\chromedriver.exe");
        DesiredCapabilities capabilities = DesiredCapabilities.chrome();
        ChromeOptions options = new ChromeOptions();
//this statement will not show any warnings when you launch chrome
        options.addArguments("test-type");
//start a maximized window
        options.addArguments("start-maximized");
//this allows you to use a user profile
        capabilities.setCapability("chrome.binary", "C:\\Server\\templates\\selenium\\chromedriver.exe");
        capabilities.setCapability(ChromeOptions.CAPABILITY, options);
        WebDriver driver = new ChromeDriver(capabilities);
```

and returned error

```
Exception in thread "main" java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkState(ZLjava/lang/String;Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)V
	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:111)
	at org.openqa.selenium.chrome.ChromeDriverService.access$000(ChromeDriverService.java:32)
	at org.openqa.selenium.chrome.ChromeDriverService$Builder.findDefaultExecutable(ChromeDriverService.java:137)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:302)
	at org.openqa.selenium.chrome.ChromeDriverService.createDefaultService(ChromeDriverService.java:88)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:146)
```

How to solve this problem?

**Comment by drno-reg** [comment]
Change version package guava.
Upgrade to guava-21.0.jar and this problem resoved.

**Comment by v-antsharma** [comment]
worked for me

**Comment by ashutoshdobhal** [comment]
download updated guava jar  https://github.com/google/guava/wiki/Release22

It works for me..

**Comment by nirojan0930** [comment]
I have tried the latest version of the guava (22.0) but it's not working for me. I have got the same error again.

**Comment by asifraaja** [comment]
I am getting this error. I have updated guava to 21 but still getting this error.

My code is :
driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);

       FluentWait<WebDriver> wait = new FluentWait<WebDriver>(driver)   
      .withTimeout(30, TimeUnit.SECONDS)   
      .pollingEvery(5, TimeUnit.SECONDS)  
      .ignoring(NoSuchElementException.class);

        wait.until(new Function<WebDriver,WebElement>() {   
        	@Override
        	public WebElement apply(WebDriver driver) {   
          return welement;   
      }
        });


Error:

java.lang.NoSuchMethodError: org.openqa.selenium.support.ui.FluentWait.until(Lcom/google/common/base/Function;)Ljava/lang/Object;
	at com.asda.qa.utility.FluentWaitImplicit.waitForElementToLoad(FluentWaitImplicit.java:155)
	at com.asda.qa.bcc.page.BCCPage.login(BCCPage.java:301)
	at com.asda.qa.linksave.LinksaveSetUpTest.createLinksaveManaullyPostXML(LinksaveSetUpTest.java:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:696)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:132)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:230)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:76)

**Comment by digz6666** [comment]
@asifraaja Use guava version 22.0

**Comment by vitekes** [comment]
I've got same error. I tried to use guava 22 and 23.

```try {
    return new InternetExplorerDriver();
} catch(NoSuchMethodError e) {
    throw new DriverException("Can not start driver", e);
}
```

java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkArgument(ZLjava/lang/String;I)V

**Comment by tis-the-flower** [comment]
Upgraded guava to version 22. It fixed the issue.

**Comment by mirro0707** [comment]
conflict with some jars

**Comment by ritesh2689** [comment]
Tried with both Guava 21 and Guava 22 , still getting the same issue. Can someone help here.

**Comment by ritesh2689** [comment]
java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkArgument(ZLjava/lang/String;Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)V
	at org.openqa.selenium.firefox.Preferences.checkPreference(Preferences.java:224)
	at org.openqa.selenium.firefox.Preferences.setPreference(Preferences.java:154)
	at org.openqa.selenium.firefox.Preferences.setPreference(Preferences.java:125)
	at org.openqa.selenium.firefox.Preferences.readDefaultPreferences(Preferences.java:104)
	at org.openqa.selenium.firefox.Preferences.<init>(Preferences.java:65)
	at org.openqa.selenium.firefox.FirefoxProfile.<init>(FirefoxProfile.java:81)
	at org.openqa.selenium.firefox.FirefoxProfile.<init>(FirefoxProfile.java:71)
	at org.openqa.selenium.firefox.FirefoxProfile.<init>(FirefoxProfile.java:60)
	at org.openqa.selenium.firefox.FirefoxDriver.toExecutor(FirefoxDriver.java:139)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:120)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:98)
	at com.snapdeal.payments.qe.checkout.configuration.DriverFactory.setDriver(DriverFactory.java:185)
	at com.snapdeal.payments.qe.checkout.configuration.DriverFactory.setDriver(DriverFactory.java:174)
	at com.snapdeal.payments.qe.checkout.utils.TestBase.beforeTest(TestBase.java:437)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:552)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:215)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:636)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:132)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:230)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:76)

**Comment by gauravbansal5464** [comment]
I am also getting the same error as you are getting @ritesh2689 can anybody please help me here. I have also used the guava 23 and guava 22 version.but still getting the same error.

**Comment by jconsultant88** [comment]
I got similar issue when using maven, selenium and juice frameworks,Ii resolved it by excluding the older version 'guava' in my class path. Maven downloaded transitive dependencies of both 'selenium-java' and 'guice' and ended up keeping guava23.0 and guava19 in the class path.
Hope fully this might give a clue @gauravbansal5464

**Comment by vincentwlau** [comment]
I ran into a similar problem, then I ran "mvn dependency:tree" to see if there was another module (e.g. google-api-client-appengine) having dependency on guava-jdk5.  Apparently guava-jdk5 is interfering with guava.jar.  Depending which jar was loaded first, you may or may not run into NoSuchMethodError.

In the pom.xml, I added an explicit guava 22.0 as dependency and added an exclusion
<dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>22.0</version>
 </dependency>
    <dependency>
      <groupId>com.google.api-client</groupId>
      <artifactId>google-api-client-appengine</artifactId>
      <version>1.23.0</version>
      <exclusions>
        <exclusion>
          <groupId>com.google.guava</groupId>
          <artifactId>guava-jdk5</artifactId>
        </exclusion>
      </exclusions>
    </dependency>

**Comment by nicoleliu125** [comment]
i have the same error with build.gradle dependencies like:
// https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium
compile group: 'org.seleniumhq.selenium', name: 'selenium', version: '2.0b1'

     // https://mvnrepository.com/artifact/org.seleniumhq.selenium/htmlunit-driver
compile group: 'org.seleniumhq.selenium', name: 'htmlunit-driver', version: '2.29.0'

// https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-chrome-driver
compile group: 'org.seleniumhq.selenium', name: 'selenium-chrome-driver', version: '3.8.1'

// https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-remote-driver
compile group: 'org.seleniumhq.selenium', name: 'selenium-remote-driver', version: '3.8.1'

// https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-server-standalone
compile group: 'org.seleniumhq.selenium', name: 'selenium-server-standalone', version: '2.39.0'

// https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-firefox-driver
compile group: 'org.seleniumhq.selenium', name: 'selenium-firefox-driver', version: '3.8.1'

// https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-java
compile group: 'org.seleniumhq.selenium', name: 'selenium-java', version: '3.8.1'
// https://mvnrepository.com/artifact/org.seleniumhq.selenium/selenium-server
compile group: 'org.seleniumhq.selenium', name: 'selenium-server', version: '3.8.1'

// https://mvnrepository.com/artifact/com.google.guava/guava
compile group: 'com.google.guava', name: 'guava', version: '24.0-jre'

there is guava dependency, and i don't know how to make it works. can anybody help?

**Comment by dmatej** [comment]
Note that for example embedded payara contains guava too - then the order of dependendies does matter.

**Comment by KrishnaDholariya** [comment]
I am getting this error. 

My code is:
package CommonFunction;

import org.testng.annotations.Test;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.BeforeTest;

import java.util.concurrent.TimeUnit;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Wait;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.AfterTest;

public class CommonAction {
	 WebDriver driver;
	 

  @Test
  public void LoginPage() {
	  openBrowser();
	  CommonLocators comlocator = new CommonLocators();
	 WebDriverWait wait = new WebDriverWait(driver, 10);
	  WebElement element = Wait.until(ExpectedConditions.elementToBeClickable(By.name(comlocator.Username_name)));  
	 
  }
  
  @BeforeTest
	  public void openBrowser() {
			
			System.setProperty("webdriver.gecko.driver", "./drivers/geckodriver.exe");
			driver = new FirefoxDriver(); 
			driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
			driver.get("http://intranet.gateway.local/default.aspx?fuseaction=login.login&ReturnUrl=%2fDashboard.aspx");	
			driver.manage().window().maximize();
			
  }
  
  @AfterTest
  public void afterMethod() {
	  driver.close();
  }

}


Error: 

[RemoteTestNG] detected TestNG version 6.11.0
FAILED CONFIGURATION: @BeforeMethod openBrowser
java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkState(ZLjava/lang/String;Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)V
	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:124)
	at org.openqa.selenium.chrome.ChromeDriverService.access$000(ChromeDriverService.java:32)
	at org.openqa.selenium.chrome.ChromeDriverService$Builder.findDefaultExecutable(ChromeDriverService.java:137)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:339)
	at org.openqa.selenium.chrome.ChromeDriverService.createDefaultService(ChromeDriverService.java:88)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:123)
	at CommonFunction.CommonAction.openBrowser(CommonAction.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:523)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:224)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:599)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)

SKIPPED: LoginPage
java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkState(ZLjava/lang/String;Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)V
	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:124)
	at org.openqa.selenium.chrome.ChromeDriverService.access$000(ChromeDriverService.java:32)
	at org.openqa.selenium.chrome.ChromeDriverService$Builder.findDefaultExecutable(ChromeDriverService.java:137)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:339)
	at org.openqa.selenium.chrome.ChromeDriverService.createDefaultService(ChromeDriverService.java:88)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:123)
	at CommonFunction.CommonAction.openBrowser(CommonAction.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:523)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:224)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:599)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)


===============================================
    Default test
    Tests run: 1, Failures: 0, Skips: 1
    Configuration Failures: 1, Skips: 0
===============================================


===============================================
Default suite
Total tests run: 1, Failures: 0, Skips: 1
Configuration Failures: 1, Skips: 0
===============================================
 

Maven dependency:

<dependency>
    <groupId>org.seleniumhq.selenium</groupId>
    <artifactId>selenium-java</artifactId>
    <version>3.11.0</version>
</dependency>
<dependency>
        <groupId>org.seleniumhq.selenium</groupId>
        <artifactId>selenium-support</artifactId>
        <version>3.0.1</version>
</dependency>
<dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>21.0</version>
</dependency>

**Comment by melihsancak** [comment]
same error mate

**Comment by shehzadafahad** [comment]
try with selenium 2.45 and chrome driver version accordingly because selenium 2.45 version using Guava 21. It resolved my issue.

**Comment by HarikaLadi** [comment]
To resolve this issue Remove if you have added any jar files under referenced Library as it creates conflicts. 
Remove the jar it should work fine.

**Comment by karippery** [comment]
```
C:\devapps\java\java\bin\java.exe -agentlib:jdwp=transport=dt_socket,address=127.0.0.1:55542,suspend=y,server=n -Dselenide.browser=chrome -Dwebdriver.chrome.driver=D:/superskill/Workspace/users/johnk/chromedriver.exe -Didea.test.cyclic.buffer.size=1048576 -javaagent:C:\devapps\intellij\community\plugins\Groovy\lib\agent\gragent.jar -javaagent:C:\devapps\intellij\community\lib\rt\debugger-agent.jar=file:/C:/Users/john/AppData/Local/Temp/233/capture497.props -Dfile.encoding=UTF-8 -classpath C:\Users\john\AppData\Local\Temp\233\classpath261528886.jar com.intellij.rt.execution.junit.JUnitStarter -ideVersion5 -junit4 test.java.com.haulmont.masquerade.LoginWindowTest,loginTest
Connected to the target VM, address: '127.0.0.1:55542', transport: 'socket'
Dec 12, 2018 4:25:36 PM com.codeborne.selenide.drivercommands.LazyDriver getAndCheckWebDriver
INFO: No webdriver is bound to current thread: 1 - let's create a new webdriver
Disconnected from the target VM, address: '127.0.0.1:55542', transport: 'socket'

java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkState(ZLjava/lang/String;Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)V

	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:134)
	at org.openqa.selenium.chrome.ChromeDriverService.access$000(ChromeDriverService.java:35)
	at org.openqa.selenium.chrome.ChromeDriverService$Builder.findDefaultExecutable(ChromeDriverService.java:159)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:355)
	at org.openqa.selenium.chrome.ChromeDriverService.createDefaultService(ChromeDriverService.java:94)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:157)
	at com.codeborne.selenide.webdriver.ChromeDriverFactory.create(ChromeDriverFactory.java:23)
	at com.codeborne.selenide.webdriver.WebDriverFactory.lambda$createWebDriver$1(WebDriverFactory.java:55)
	at java.util.Optional.map(Optional.java:215)
	at com.codeborne.selenide.webdriver.WebDriverFactory.createWebDriver(WebDriverFactory.java:55)
	at com.codeborne.selenide.drivercommands.CreateDriverCommand.createDriver(CreateDriverCommand.java:38)
	at com.codeborne.selenide.drivercommands.LazyDriver.createDriver(LazyDriver.java:93)
	at com.codeborne.selenide.drivercommands.LazyDriver.getAndCheckWebDriver(LazyDriver.java:87)
	at com.codeborne.selenide.SelenideDriver.getAndCheckWebDriver(SelenideDriver.java:143)
	at com.codeborne.selenide.drivercommands.Navigator.navigateTo(Navigator.java:61)
	at com.codeborne.selenide.drivercommands.Navigator.open(Navigator.java:24)
	at com.codeborne.selenide.SelenideDriver.open(SelenideDriver.java:59)
	at com.codeborne.selenide.Selenide.open(Selenide.java:39)
	at test.java.com.haulmont.masquerade.LoginWindowTest.loginTest(LoginWindowTest.java:16)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)


Process finished with exit code -1
```



**build.gradle**
```
  compile group: 'com.google.guava', name: 'guava', version: '11.0.2'
  compile 'com.google.guava:guava:22.0'
compile group: 'org.seleniumhq.selenium', name: 'selenium-chrome-driver', version: '2.23.0'
        testCompile group: 'org.seleniumhq.selenium', name: 'selenium-java', version: '3.141.59'
        compile group: 'org.seleniumhq.selenium', name: 'selenium-htmlunit-driver', version: '2.52.0'
        compile group: 'org.seleniumhq.selenium', name: 'selenium-server', version: '3.141.59'

```

**Comment by kajoldoss** [comment]
getting the same error after adding guava dependencies also. Please help me. I am using maven + TestNG

**Comment by kamranayub** [comment]
Just wanted to post here in case anyone else comes to this thread from Google like I did. For whatever reason, I needed to run with `sudo`. I was having issues using the npm `selenium-standalone` package and running:

    /node_modules/selenium-standalone/bin/selenium-standalone start

And it would show that error. What fixed it was running with sudo

    sudo /node_modules/selenium-standalone/bin/selenium-standalone start

I don't think I needed to do this before but suddenly it's the only way it works now.
